extensions/etcdsnapshot/etcdsnapshot.go

```diff
@@ -260,18 +283,40 @@ func RestoreRKE1Snapshot(client *rancher.Client, clusterName string, snapshotRes
 }
 
 // RestoreRKE2K3SSnapshot is a helper function to restore a snapshot on an RKE2 or k3s cluster. Returns error if any.
-func RestoreRKE2K3SSnapshot(client *rancher.Client, clusterName string, snapshotRestore *rkev1.ETCDSnapshotRestore, initialControlPlaneValue, initialWorkerValue string) error {
-	clusterObject, existingSteveAPIObject, err := clusters.GetProvisioningClusterByName(client, clusterName, fleetNamespace)
+func RestoreRKE2K3SSnapshot(client *rancher.Client, snapshotRestore *rkev1.ETCDSnapshotRestore, cluster *apisV1.Cluster) error {
+	clusterObject, existingSteveAPIObject, err := clusters.GetProvisioningClusterByName(client, cluster.Name, fleetNamespace)
 	if err != nil {
 		return err
 	}
 
 	clusterObject.Spec.RKEConfig.ETCDSnapshotRestore = snapshotRestore
-	clusterObject.Spec.RKEConfig.UpgradeStrategy.ControlPlaneConcurrency = initialControlPlaneValue
-	clusterObject.Spec.RKEConfig.UpgradeStrategy.WorkerConcurrency = initialWorkerValue
+	clusterObject.Spec.RKEConfig.UpgradeStrategy.ControlPlaneConcurrency = cluster.Spec.RKEConfig.UpgradeStrategy.ControlPlaneConcurrency
+	clusterObject.Spec.RKEConfig.UpgradeStrategy.WorkerConcurrency = cluster.Spec.RKEConfig.UpgradeStrategy.WorkerConcurrency
 
 	logrus.Infof("Restoring snapshot: %v", snapshotRestore.Name)
-	_, err = client.Steve.SteveType(ProvisioningSteveResouceType).Update(existingSteveAPIObject, clusterObject)
+	updatedCluster, err := client.Steve.SteveType(ProvisioningSteveResouceType).Update(existingSteveAPIObject, clusterObject)
+	if err != nil {
+		return err
+	}
+
+	err = kwait.PollUntilContextTimeout(context.TODO(), 500*time.Millisecond, defaults.FifteenMinuteTimeout, true, func(ctx context.Context) (done bool, err error) {
+		clusterResp, err := client.Steve.SteveType(ProvisioningSteveResouceType).ByID(updatedCluster.ID)
+		if err != nil {
+			return false, err
+		}
+
+		clusterStatus := &apisV1.ClusterStatus{}
+		err = v1.ConvertToK8sType(clusterResp.Status, clusterStatus)
+		if err != nil {
+			return false, err
+		}
+
+		if clusterResp.ObjectMeta.State.Name == active {
```

Comment:
this is different than rke1 with the `wait on wait` comment from that function. Is that intentional?

Reply:
Yeah, this is meant to be the way it is. RKE1 wait status is different in that the cluster will show as `Active`, but the nodes are not all `Active` when that happens. v2 prov doesn't have that same issue.